### PR TITLE
Add affordance to input labels

### PR DESCRIPTION
**Pull request by neutraltone**
I noticed that the `label`s for form elements such as text inputs, radio buttons and checkboxes did not use a `pointer` cursor on hover.

This PR adds clickable `label`s for text inputs and how displays a `pointer` cursor when hovering the labels. This improves UX and accessibility one the form elements.

### CODE REVIEW

**Comment by BcRikko** [comment]
Thanks!

I'm not very famiiar with UX and accessibility. so please let me ask a question. 🙏 
Are #16 and #17 not enough? 🤔 

If possible, I don't want IDs attached to `input`.
because I want to avoid ID duplication.

Could you teach this to me? 🙇

**Comment by scottaohara** [comment]
@BcRikko, per your demo site, it looks like you're already wrapping the checkboxes and radio buttons in `label` elements, so you should be OK regarding `id`s.  `id`s are only required for accessibility if the input needs to exist outside of its associated `label`.

So if your code was like:
```html
<label>name</label><input type="checkbox">
```
you would need the `for` attribute on the `label` pointing to the `id` of the `input`.

Regarding the cursor change on `:hover`, right now #17 only affects the `span` within the `label`, so the checkbox and radio UI elements don't have a cursor change associated with them.  That should be considered a bug since everything associated with the `input` should receive this cursor change, if used.

That said, natively radio buttons, and checkboxes do not have a cursor change on `:hover`.  Links do, but not form controls.  Regardless, some people have come to expect the change, so it's one of those "technically it's not right, but people are beginning to expect it" sort of things, and if you're going to employ it into your CSS, it should be consistent :)

If you're interested, I've written quite a bit about styling native form controls and keeping them accessible.  Here are my write-ups for [checkboxes](https://scottaohara.github.io/a11y_styled_form_controls/src/checkbox/) and [radio buttons](https://scottaohara.github.io/a11y_styled_form_controls/src/radio-button/) specifically.

**Comment by BcRikko** [comment]
@scottaohara  Thanks for explaining in detail 🙇 
I can learn a lot from you! 😄 

but it's late night in Japan, so I will read your comment and a11y_styled_form_controls tomorrow. sorry... 😪

**Comment by scottaohara** [comment]
no worries @BcRikko.  the info isn't going anywhere.  good night :)

File: scss/base/generic.scss [hunk 1]
```diff
@@ -21,3 +21,7 @@ button,
 [type="submit"] {
   -webkit-appearance: none;
 }
+
+label:hover {
+  cursor: pointer;
```
[the remark this answers is not in the document]
Comment (scottaohara) [reply scss/base/generic.scss]: Scoping this to `for` won't solve the issue with the current pattern, as there are no `for` attributes. The `label` is associated with the `input` because the `input` is a child of the `label`.  

You'd have to revise the selector like:
```css
label[for]:hover,
label input:hover { cursor: pointer; }
```
And then you'd still need to account for hovering over the `span`s within these `label`s w/out `for` attributes.

[the remark this answers is not in the document]
Comment (agarzola) [reply scss/base/generic.scss]: I don’t see the value in adding the `label input:hover` selector, since that would make the _input_ get the pointer cursor, not the label itself. Using the `for` attribute as this PR suggests is the correct solution here, and I would add that the markup where the `input` is nested in the `label` should be modified to use `for`.

My suggestion here still stands: the pointer cursor should not be applied to `label` elements that do nothing for the user when clicked.

[the remark this answers is not in the document]
Comment (scottaohara) [reply scss/base/generic.scss]: Hi @agarzola,

I will re-explain for you.

As @BcRikko requested previously:
> If possible, I don't want IDs attached to `input`.

Being that `id`s are not required on an `input` if it is a child of a `label`, this is where your solution alone would not cover the present markup pattern and the desire expressed to not used `id` on inputs if not necessary.  

I'm not disagreeing with your opinion to include the `label[for]:hover` selector.  Though I tend to side with the [arguments made in this article regarding mouse cursors](https://medium.com/simple-human/buttons-shouldnt-have-a-hand-cursor-b11e99ca374b), it's more important to be consistent and ensure to not let corner cases slip through if the goal is to buck native functionality and use pointer cursors on all interactive elements.

I hope this helps you understand the value of my previous comment.

[the remark this answers is not in the document]
Comment (agarzola) [reply scss/base/generic.scss]: Hey, @scottaohara. 👋 Thanks for expanding on your previous comment.

To clarify: I do see value in your comment (and this exchange!). 😄 I just don’t see the value in making the `input` itself get the pointer cursor, which is what adding the `label input:hover` selector would do.

We’re in agreement that labels shouldn’t get pointer cursors in the first place, as well as the assertion that —conventions not withstanding— consistency is paramount. And it’s consistency in the experience that I’m trying to address with my suggestion, because a `label` _without_ a `for` attribute and no nested `input` does nothing for the user when clicked, so why should the cursor change, implying that the label is interactive?

So again, while we’re in agreement that the most appropriate answer _should be_ “let’s remove pointer cursor from `label` elements altogether”, if one must be applied then I think the only way to be as consistent as possible is to avoid the cursor change when hovering over inert elements.

I hope that clarifies my position on this.

[the remark this answers is not in the document]
Comment (agarzola) [reply scss/base/generic.scss]: You’re right. I missed that! 😄 Yes: the correct solution is to revert the work that gives labels a pointer cursor.

**Comment by scottaohara** [comment]
Hi @BcRikko 

The work done in #30 didn't touch on any of the work this PR.

tldr; probably should just close it :)

<details>
<summary>Longer explanation</summary>
<p>Per the discussion @agarzola and I were having, IMO this PR moves to solve some cursor consistency issues, but as <a href="https://github.com/BcRikko/NES.css/pull/20#discussion_r237481212">pointed out</a>, a blanket <code>label:hover</code> cursor change can potentially lead to some misleading UX if `label`s aren't used correctly. 
<p>
Furthermore, ideally the cursor for buttons and other form controls wouldn't be altered at all as the "pointer" is meant to indicate an element is a hyperlink, and not a globally "clickable" element.
</details>
